src/components/ListHeader.js

```diff
@@ -16,18 +16,35 @@ export const ListHeader = ({ data }) => {
     ).map((dataProvider, index) => ({
       id: index + 1,
       value: dataProvider,
-      selected: false
+      selected: dataProvider === queryVariables.dataProvider
     }));
 
   const [dropdownData, setDropdownData] = useState([
     {
       id: 0,
       value: '- Alle -',
-      selected: true
+      selected: !queryVariables.dataProvider
     },
     ...dataProviders
   ]);
 
+  const selectedDropdownData = dropdownData.find((entry) => entry.selected);
+
+  // https://medium.com/swlh/prevent-useeffects-callback-firing-during-initial-render-the-armchair-critic-f71bc0e03536
+  const initialRender = useRef(true);
+
+  // influence list data when changing selected dropdown value
+  // call update of the list with the selected data provider
+  useEffect(() => {
+    if (initialRender.current) {
+      initialRender.current = false;
```

Comment:
in this way we are preventing the default initialRender when useEffect is triggered

Reply:
yes, what we got from the link a few lines above: https://medium.com/swlh/prevent-useeffects-callback-firing-during-initial-render-the-armchair-critic-f71bc0e03536